plutus-benchmark/src/Plutus/Benchmark/Clausify.hs

```diff
@@ -0,0 +1,201 @@
+{-# INLINABLE tautclause #-}
+tautclause :: LRVars -> Bool
+tautclause (c,a) = [x | x <- c, x `elem` a] /= []
+
+-- form unique clausal axioms excluding tautologies
+{-# INLINABLE unicl #-}
+unicl :: [Formula] -> [LRVars]
+unicl a = foldr unicl' [] a
+          where
+          unicl' p x = if tautclause cp then x else insert cp x
+                       where
+                       cp = clause p :: LRVars
+
+{-# INLINABLE while #-}
+while :: (t -> Bool) -> (t -> t) -> t -> t
+while p f x = if p x then while p f (f x) else x
+
+{-# INLINABLE replicate #-}
+replicate :: Integer -> a -> [a]
+replicate n a = if n == 0 then []
+                else a:(replicate (n-1) a)
+
+{-# INLINABLE formula1 #-}  -- Overflow
+formula1 :: Formula  -- (a = a = a) = (a = a = a) = (a = a = a)
+formula1 = Eqv (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+               (Eqv (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+                    (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1))))
+
+{-# INLINABLE formula2 #-} -- Overflow
+formula2 :: Formula -- (a = b = c) = (d = e = f) = (g = h = i)
+formula2 = Eqv (Eqv (Sym 1) (Eqv (Sym 2) (Sym 3)))
+               (Eqv (Eqv (Sym 4) (Eqv (Sym 5) (Sym 6)))
+                    (Eqv (Sym 7) (Eqv (Sym 8) (Sym 9))))
+
+{-# INLINABLE formula3 #-}
+formula3 :: Formula  -- (a = a = a) = (a = a = a)
+formula3 = Eqv (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+               (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+
+{-# INLINABLE formula4 #-} -- One execution takes about 0.35s and 300 MB
+formula4 :: Formula  -- (a = a) = (a = a) = (a = a)
+formula4 = Eqv (Eqv (Sym 1) (Sym 1))
+               (Eqv (Eqv (Sym 1) (Sym 1))
+                    (Eqv (Sym 1) (Sym 1)))
+
+{-# INLINABLE formula5 #-}  -- One execution takes about 1.5s and 660 MB
+formula5 :: Formula  -- (a = a = a) = (a = a) = (a = a)
+formula5 = Eqv (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+               (Eqv (Eqv (Sym 1) (Sym 1))
+                    (Eqv (Sym 1) (Sym 1)))
+
+{-# INLINABLE formula5a #-}  -- One execution takes about 2s and 1 GB
+formula5a :: Formula  -- (a = b = c) = (d = e) = (f = g)
+formula5a = Eqv (Eqv (Sym 1) (Eqv (Sym 2) (Sym 3)))
+               (Eqv (Eqv (Sym 4) (Sym 5))
+                    (Eqv (Sym 6) (Sym 7)))
+
+{-# INLINABLE formula6 #-}  -- One execution takes about 11s and 5 GB
+formula6 :: Formula  -- (a = a = a) = (a = a = a) = (a = a)
+formula6 = Eqv (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+               (Eqv (Eqv (Sym 1) (Eqv (Sym 1) (Sym 1)))
+                    (Eqv (Sym 1) (Sym 1)))
+
+data StaticFormula = F1 | F2 | F3 | F4 | F5 | F5A | F6
+
+{-# INLINABLE getFormula #-}
+getFormula :: StaticFormula -> Formula
+getFormula F1  = formula1
+getFormula F2  = formula2
+getFormula F3  = formula3
+getFormula F4  = formula4
+getFormula F5  = formula5
+getFormula F5A = formula5a
+getFormula F6  = formula6
+
+{-# INLINABLE mkClausifyTerm #-}
+mkClausifyTerm :: Integer -> StaticFormula -> Term Name DefaultUni ()
+mkClausifyTerm cnt formula =
```

Comment:
There is one: `CompiledCode`. But Kenneth explicitly gets the term inside it here. Which seems okay to me since this is about creating inputs to test the machine.